authentication/managing_cloud_provider_credentials/cco-mode-sts.adoc

```diff
@@ -0,0 +1,615 @@
+= Using STS mode
+include::modules/common-attributes.adoc[]
+:context: cco-mode-sts
+toc::[]
+
+STS mode is supported for Amazon Web Services (AWS).
+
+:FeatureName: Support for AWS Secure Token Service (STS)
+include::modules/technology-preview.adoc[leveloffset=+0]
+
+In STS mode, the Cloud Credential Operator (CCO) uses AWS Secure Token Service (STS) to assign components IAM roles that provide short-term, limited-privilege security credentials.
+
+Requests for new and refreshed credentials are automated by using an AWS IAM OpenID Connect (OIDC) identity provider. {product-title} signs service account tokens that are trusted by AWS IAM, and can be projected into a pod and used for authentication. Tokens are refreshed after one hour.
+
+.STS authentication flow
+image::sts_flow.png[STS authentication flow]
+
+The implementation of STS mode changes the format of the AWS credentials root secret in the `kube-system` namespace.
+
+.AWS secret format using a mode other than STS
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: kube-system
+  name: aws-creds
+data:
+  aws_access_key_id: <base64-encoded-access-key-id>
+  aws_secret_access_key: <base64-encoded-secret-access-key>
+----
+
+.AWS secret format using STS mode
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: kube-system
+  name: aws-creds
+data:
+  role_name: <operator-role-name> <1>
+  web_identity_token_file: <path-to-token> <2>
+----
+<1> The IAM role for the cluster
++
+<needinfo> Is this accurate/informative? or is there a better way to describe it?
+<2> The path to the service account tokens.
+
+[id="sts-mode-installing"]
+== Installing an {product-title} cluster configured for STS mode
+
+In {product-title} version 4.7, installing a cluster that is configured to use the CCO in STS mode requires the following steps:
+
+. xref:#sts-mode-installing-preparing[Preparing the installer]
+. Creating AWS resources xref:#sts-mode-installing-config-preflight[using the `sts-preflight` tool] or xref:#sts-mode-installing-config-manual[manually]
+. xref:#sts-mode-installing-running-installer[Running the installer]
+. xref:#sts-mode-installing-verifying[Verifying the installation]
+
+[NOTE]
+====
+<needinfo>
+Are they basically following the link:https://docs.openshift.com/container-platform/4.6/installing/installing_aws/installing-aws-customizations.html#installation-initializing_installing-aws-customizations[Installing a cluster on AWS with customizations] procedure here, with some specific differences for STS (and of course the addition of the preflight stuff)?
+====
+
+
+[id="sts-mode-installing-preparing"]
+=== Preparing the installer
+
+.Prerequisites
+* Obtain the pull secret for your cluster.
+
+.Procedure
+
+. Set the `$RELEASE_IMAGE` environment variable to use a sufficiently new {product-title} release:
++
+[source,terminal]
+----
+<needinfo>
+- Need information about how to do this (commands, output if applicable)
```

Comment:
Once you have the openshfit-install binary, you can run `openshift-install version` and it will show you the release image associated with the specific version of the installer that is being run.
It comes out in a non-human-friendly format though (it doesn't actually encode the version string into the release image URL/text:

```
[jdiaz@minigoomba os-install-4.6.9]$ ./openshift-install version
./openshift-install 4.6.9
built from commit a48ad4a15b42102d1747d2f5f3b635deffb950b5
release image quay.io/openshift-release-dev/ocp-release@sha256:43d5c84169a4b3ff307c29d7374f6d69a707de15e9fa90ad352b432f77c0cead
```

But if you browse to https://quay.io/repository/openshift-release-dev/ocp-release?tab=tags and search/filter for `4.6.9-x86_64` you will see that the SHA256 for the 4.6.9-x86_64 matches the release image printed out by the installer.

![image](https://user-images.githubusercontent.com/4999586/108235538-e7ab1b80-7113-11eb-918e-c3bd6bac3749.png)

Reply:
I _think_ I resolved the bit with these first steps by pulling from the manually creating IAM content. Hopefully refreshed PR will get this cleaned up.